rust/src/utils/testing.rs

```diff
@@ -42,3 +51,90 @@ pub fn generate_random_array(n: usize) -> Float32Array {
             .collect::<Vec<f32>>(),
     )
 }
+
+pub async fn create_file(
+    path: &std::path::Path,
+    mode: WriteMode,
+    dims: i32,
+    num_rows: i32,
+    batch_size: i32,
+    seed: [u8; 32],
+) {
```

Comment:
It would be nice to take the parameters as an options struct. Something like:

```rust
pub struct CreateFileOptions {
    dims: i32,
    num_rows: i32,
    batch_size: i32,
    seed: [u8; 32],
}

pub async fn create_file(
    path: &std::path::Path,
    mode: WriteMode,
    options: &CreateFileOptions
) { ...
```

Then invocations looks cleaner:

```rust
create_file(
    std::path::Path::new(path),
    WriteMode::Overwrite,
    &CreateFileOptions {
        dims: 3,
        num_rows: 200,
        batch_size: 4,
        seed: [0; 32]
    }
)
```

Then if you also defined `Default` for it:

```rust
impl Default for CreateFileOptions {
    fn default() -> Self {
        Self {
            dims: 3,
            num_rows: 200,
            batch_size: 4,
            seed: [0; 32]
        }
    }
}
```

Then you can leave out arguments with:

```rust
create_file(
    std::path::Path::new(path),
    WriteMode::Overwrite,
    &CreateFileOptions {
        dims: 3,
        ..Default::default()
    }
)
```